src/ingress.ts

```diff
@@ -270,9 +270,20 @@ export class IngressBackend {
    * A Kubernetes `Service` to use as the backend for this path.
    * @param serv The service object.
    */
-  public static fromService(serv: service.Service, options: ServiceIngressBackendOptions = {}) {
-    if (serv.ports.length === 0) {
-      throw new Error('service does not expose any ports');
+  public static fromService(serv: service.IService, options: ServiceIngressBackendOptions = {}) {
+    // For imported services, ports might be undefined or empty
+    // In that case, we must rely on the port provided in options
+    if (!serv.ports || serv.ports.length === 0) {
+      if (options.port !== undefined) {
+        return new IngressBackend({
+          service: {
+            name: serv.name,
+            port: { number: options.port },
```

Comment:
I understand where you're going with this, but it causes `options.port` to have multiple designations in the function. Note that `options.port` is currently only used as _disambiguator_ in case the service itself exposes more than one port, as evident by:

https://github.com/cdk8s-team/cdk8s-plus/blob/5963ee2714d1f45b9cb7fa738f93afcbb6207b8d/src/ingress.ts#L294-L296

With this change though, it suddenly becomes an _override_, just for this specific code path. 

Instead, I prefer we allow the user to specify a port during service import. See below.